src/java/org/apache/cassandra/db/commitlog/CommitLogSegment.java

```diff
@@ -499,13 +503,12 @@ void waitForFinalSync()
         }
     }
 
-    void waitForSync(int position, Timer waitingOnCommit)
+    void waitForSync(int position)
     {
         while (lastSyncedOffset < position)
         {
-            WaitQueue.Signal signal = waitingOnCommit != null ?
-                                      syncComplete.register(waitingOnCommit.time()) :
-                                      syncComplete.register();
```

Comment:
A general question... why does `AbstractCommitLogService#awaitSyncAt` (called by `PeriodicCommitLogService`) still register the timer context with the syncComplete?

Reply:
`PeriodicCommitLogService` doesn't block writes under normal conditions, so I left that behavior alone. The big motivating factor for this change was making sure that when "batch" or "group" CL are in use, the metric data points correspond to single writes (and vise-versa). It probably wouldn't be harmful to do it for "periodic" as well, but it also doesn't buy much.